### DISCUSSION TITLE
How to delete internal topic

### DISCUSSION THREAD

**Discussion by kaykhan** [provectus/kafka-ui#4101] [Q&A] [answered]
When we create the connector msk, we need to delete the old internal topics.

But currently kafka-ui show the button grayed out, we can delete these topics manually via the kafka-topics but how do we delete them using the kafka ui. 

![image](https://github.com/provectus/kafka-ui/assets/5202527/ba7a4cf2-0de1-4ff0-84b1-b59b110bcc36)

**Comment by Haarolean** [comment, marked as answer]
Hi, you can't delete them because they're internal. However, you can modify the internal prefix to an empty string to make it work. Change `kafka_internalTopicPrefix`.

**Reply by kaykhan** [reply]
Okay, is it possible to give a specific user the permission to perform this action? I dont want all users to be able to perform this.

**Reply by Haarolean** [reply]
@kaykhan yes, with [RBAC](https://docs.kafka-ui.provectus.io/configuration/rbac-role-based-access-control):
```
        - resource: topic
          value: "_.*"
          actions: delete
```
